dev/AppNotifications/AppNotificationUtility.cpp

```diff
@@ -232,61 +246,61 @@ void Microsoft::Windows::AppNotifications::Helpers::RegisterAssets(std::wstring
+        LOG_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchDisplayNameResource, &propVariantDisplayName));
+        if (propVariantDisplayName.vt == VT_LPWSTR && propVariantDisplayName.pwszVal != nullptr)
+        {
+            displayName = propVariantDisplayName.pwszVal;
+        }
+        else if (displayName.empty())
+        {
+            displayName = SetDisplayNameBasedOnProcessName();
+        }
 
-        displayName.erase(displayName.find_first_of(L".")); // Remove file extension
-    }
+        wil::unique_prop_variant propVariantIcon;
+        THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchIconResource, &propVariantIcon));
+        THROW_HR_IF_MSG(E_UNEXPECTED, propVariantIcon.vt == VT_EMPTY, "You must specify a valid image filepath before calling Register().");
 
-    // Retrieve the icon
-    std::wstring iconFilePath{};
+        iconFilePath = propVariantIcon.pwszVal;
 
-    wil::unique_prop_variant propVariantIcon;
-    // Throw in case of failure, since Icon is mandatory and we don't have a fallback!
-    THROW_IF_FAILED(propertyStore->GetValue(PKEY_AppUserModel_RelaunchIconResource, &propVariantIcon));
-    THROW_HR_IF_MSG(E_UNEXPECTED, propVariantIcon.vt == VT_EMPTY, "You must specify an app icon before calling Register().");
+        // Icon filepaths from Shell APIs have this format: <filepath>,-<index>,
+        // since .ico files can have multiple icons in the same file.
+        // NotificationController doesn't seem to support such format, so let it take the first icon by default.
+        auto iteratorForCommaDelimiter{ iconFilePath.find_first_of(L",") };
+        if (iteratorForCommaDelimiter != std::wstring::npos) // It may or may not have an index, which is fine.
+        {
+            iconFilePath.erase(iteratorForCommaDelimiter);
+        }
 
-    iconFilePath = propVariantIcon.pwszVal;
+        auto iteratorForFileExtension{ iconFilePath.find_first_of(L".") };
+        THROW_HR_IF_MSG(E_UNEXPECTED, iteratorForFileExtension == std::wstring::npos, "You must provide a valid filepath as the app icon.");
 
-    // Icon filepaths from Shell APIs have this format: <filepath>,-<index>,
-    // since .ico files can have multiple icons in the same file.
-    // NotificationController doesn't seem to support such format, so let it take the first icon by default.
-    auto iteratorForCommaDelimiter = iconFilePath.find_first_of(L",");
-    if (iteratorForCommaDelimiter != std::wstring::npos) // It may or may not have an index, which is fine.
+        std::wstring iconFileExtension{ iconFilePath.substr(iteratorForFileExtension) };
+        THROW_HR_IF_MSG(E_UNEXPECTED, iconFileExtension != L".ico" && iconFileExtension != L".png",
+            "You must provide a supported file extension as the icon (.ico or .png).");
+    }
+    else
     {
-        iconFilePath.erase(iteratorForCommaDelimiter);
+        displayName = SetDisplayNameBasedOnProcessName();
```

Comment:
Bug: Set the displayname from processName at the beginning itself and remove this else clause.